cpp/src/trajectory/trajectory_bounding_boxes.cu

```diff
@@ -66,40 +64,19 @@ struct dispatch_element {
     cols.push_back(
       cudf::make_numeric_column(type, num_trajectories, cudf::mask_state::UNALLOCATED, stream, mr));
 
-    auto points = thrust::make_zip_iterator(thrust::make_tuple(
-      x.begin<Element>(), y.begin<Element>(), x.begin<Element>(), y.begin<Element>()));
-
-    auto bboxes = thrust::make_zip_iterator(
-      thrust::make_tuple(cols.at(0)->mutable_view().begin<Element>(),  // bbox_x1
-                         cols.at(1)->mutable_view().begin<Element>(),  // bbox_y1
-                         cols.at(2)->mutable_view().begin<Element>(),  // bbox_x2
-                         cols.at(3)->mutable_view().begin<Element>())  // bbox_y2
-    );
-
-    thrust::fill(rmm::exec_policy(stream),
-                 bboxes,
-                 bboxes + num_trajectories,
-                 thrust::make_tuple(std::numeric_limits<Element>::max(),
-                                    std::numeric_limits<Element>::max(),
-                                    std::numeric_limits<Element>::min(),
-                                    std::numeric_limits<Element>::min()));
-
-    thrust::reduce_by_key(
-      rmm::exec_policy(stream),         // execution policy
-      object_id.begin<int32_t>(),       // keys_first
-      object_id.end<int32_t>(),         // keys_last
-      points,                           // values_first
-      thrust::make_discard_iterator(),  // keys_output
-      bboxes,                           // values_output
-      thrust::equal_to<int32_t>(),      // binary_pred
-      [] __device__(auto a, auto b) {   // binary_op
-        Element x1, y1, x2, y2, x3, y3, x4, y4;
-        thrust::tie(x1, y1, x2, y2) = a;
-        thrust::tie(x3, y3, x4, y4) = b;
-        return thrust::make_tuple(
-          min(min(x1, x2), x3), min(min(y1, y2), y3), max(max(x1, x2), x4), max(max(y1, y2), y4));
-      });
+    auto points_begin = cuspatial::make_vec_2d_iterator(x.begin<T>(), y.begin<T>());
+
+    auto bbox_mins  = cuspatial::make_vec_2d_output_iterator(cols.at(0)->mutable_view().begin<T>(),
+                                                            cols.at(1)->mutable_view().begin<T>());
+    auto bbox_maxes = cuspatial::make_vec_2d_output_iterator(cols.at(2)->mutable_view().begin<T>(),
+                                                             cols.at(3)->mutable_view().begin<T>());
 
+    trajectory_bounding_boxes(object_id.begin<std::int32_t>(),
+                              object_id.end<std::int32_t>(),
```

Comment:
Should we use `cudf::size_type` here?

Reply:
I guess so. But the runtime check on line 117 is `cudf::type_id::INT32`. How do we ensure that it is `cudf::size_type`? What if size_type changes?

Reply:
I believe the type check should be `... == type_to_id<cudf::size_type>()`.